operator/README.md

```diff
@@ -1,2 +1,92 @@
-# pmem-csi-operator
-Operator for PMEM-CSI
+# PMEM-CSI-Operator
+
+Kubernetes operator for [PMEM-CSI](https://github.com/intel/pmem-csi) driver deployment. Using this operator one can easily deploy and manage PMEM-CSI driver in a Kubernetes cluster.
+
+The PMEM-CSI operator is based on the CoreOS [operator-sdk](https://github.com/operator-framework/operator-sdk) tools and APIs.
+
+## PMEM-CSI Deployment custom resource definition(CRD)
+
+Current PMEM-CSI Deployment object supports below API:
+
+### Deployment
+
+|Field | type | Description |
+|---|---|---|
+| apiVersion | string  | `pmem-csi.intel.com/v1alpha1`|
+| kind | string | `Deployment`|
+| metaData | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Standard objects metadata. |
+| spec | [DeploymentSpec](#deployment-spec) | Specification of the desired behavior of the deployment |
+
+### DeployemtSpec
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| image | ImageInfo | PMEM-CSI driver container image information | `intel/pmem-csi-driver:canary` |
+| logLevel | integer | PMEM-CSI driver logging level | 3 |
+| controller | CotrollerDriver | configuration of PMEM-CSI driver running as controller | |
+| node | NodeDriver | configuration of PMEM-CSI driver running at compute node | |
+
+### ControllerDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| provisionerImager | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-provisioner:v1.2.1` |
+| resources | [ResourceRequirements](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | describes the compute resource requirements for controller pod |
+
+### NodeDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| deviceMode | string | Device management mode to use. Supports one of `lvm` or `direct` | `lvm`
+| registrarImage | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-node-driver-registrar:v1.1.0` |
+| resources | [ResourceList](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | |
+
+### ImageInfo
+
+|Field | type | Description | Default value |
+|---|---|--|---|
+| registry | string | Docker image registry |
+| name | string | Container image name |
+| tag | string | Container image tag |
+| pullPolicy | string | Image pull policy. either one of `Always`, `Never`, `IfNotPresent` | `IfNotPresent` |
+
+## Pre-requisites
+
+### Build operator image
+
+Build PMEM-CSI operator image using the provided [Dockerfile](https://github.com/intel/pmem-csi/tree/devel/operator/build/Dockerfile) and push the image to local docker registry.
+
+```sh
+docker build --pull -t pmem-csi-operator:canary  --label revision=$(VERSION) -f operator/build/Dockerfile .
```

Comment:
Have you considered just adding the operator to the existing image?

Not having to maintain and publish another image is going to be simpler. The downside is that one additional file will be put onto each node where PMEM-CSI runs.

I think I prefer to start simple (i.e. add to existing image) and later optimize for overall size if needed.

Reply:
> Have you considered just adding the operator to the existing image?

No.

I felt like this deserves its own image :-)

Reply:
So considering the pros and cons, what do you prefer and why?

Reply:
Discussed just now, let's use one image.

Once we have something working we can discuss whether we want to optimize for size by splitting into two images.

Reply:
Please remove  https://github.com/avalluri/pmem-CSI/blob/dc89540733bd126ac20924590340e6989cbfb5bf/operator/build/Dockerfile#L1-L15

This is not an image that we would be able to ship anyway (`FROM registry.access.redhat.com/ubi8/ubi-minimal:latest`...)

Reply:
> This is not an image that we would be able to ship anyway (`FROM registry.access.redhat.com/ubi8/ubi-minimal:latest`...)

This is what PR #494 about :-)  https://github.com/intel/pmem-csi/pull/494/files#diff-04ea00daa9ec469d5b78e40cf8e7649a

But anyway we decided to ship both operator and driver with a single image, so we might not need this Dockerfile in this case.

Reply:
Exactly, so let's get rid of it.